### PR TITLE
Better retry logic for flaky end-to-end tests

### PR DESCRIPTION
**Pull request by tisnik**
## Description

Better retry logic for flaky end-to-end tests
- it is possible to specify error type
- it is possible to specify which text must be included in error message

## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

### CODE REVIEW

File: tests/e2e/test_decorators.py [hunk 1]
```diff
@@ -4,22 +4,45 @@
 from functools import wraps
 
 
-def retry(max_attempts=3, wait_between_runs=5):
-    """Construct decorator that allows to retry running selected test."""
+def retry(
+    max_attempts=3,
+    wait_between_runs=5,
+    error_type=AssertionError,
```
Comment (onmete) [inline tests/e2e/test_decorators.py:10]: on_error/on_errors?

File: tests/e2e/test_decorators.py [hunk 2]
```diff
@@ -4,22 +4,45 @@
...
+    expected_error_message="",
+):
+    """Construct decorator that allows to retry running selected test.
+
+    Args:
+        max_attempts: how many times the test will be rerun in case of failure
+        wait_between_runs: time duration between consecutive test runs
+        error_type: expected error type (other errors will mark test failure immediatelly)
+        expected_error_message: text that must be contained in error messge
```
Comment (onmete) [inline tests/e2e/test_decorators.py:19]: mess**a**ge

Comment (tisnik) [reply tests/e2e/test_decorators.py]: fixed

File: tests/e2e/test_decorators.py (hunk 1 above)
Comment (tisnik) [reply tests/e2e/test_decorators.py]: fixed @onmete

File: tests/e2e/test_decorators.py (hunk 2 above)
Comment (tisnik) [reply tests/e2e/test_decorators.py]: fixed @onmete

File: tests/e2e/test_decorators.py [hunk 3]
```diff
@@ -4,22 +4,45 @@
...
+        decorated test function
+    """
 
     def retry_test_decorator(test_function):
         @wraps(test_function)
         def wrapper(*args, **kwargs):
             retry_count = 1
 
-            while retry_count < max_attempts:
+            while retry_count <= max_attempts:
```
Comment (bparees) [inline tests/e2e/test_decorators.py:31]: since you make one final attempt at invoking the test function outside the loop, i think `<` was correct?

Comment (tisnik) [reply tests/e2e/test_decorators.py:31]: hmm yes. I was thinking the opposite - how much times to retry the test after failure, not the overall number of test runs. But can change it to `<` if you prefer (have no strong opinion).

Comment (bparees) [reply tests/e2e/test_decorators.py:31]: well you called the argument "max attempts" and not "max retries", so i'd enforce it as "maximum number of times we will invoke the test"

(I suspect that even max retries is often interpreted by people to mean max attempts/tries, not "max attempts after the first one")

it hardly matters overall, but we might as well get it right while we're looking at it.

**Pull request merged**